src/compile/legend/properties.ts

```diff
@@ -17,8 +25,39 @@ export function values(legend: Legend, fieldDef: TypedFieldDef<string>) {
   return undefined;
 }
 
-export function defaultSymbolType(mark: Mark) {
-  return mark === 'line' ? 'stroke' : 'circle';
+export function defaultSymbolType(
+  mark: Mark,
+  channel: Channel,
+  shapeChannelDef:
+    | FieldDefWithCondition<MarkPropFieldDef<string>, Value>
+    | ValueDefWithCondition<MarkPropFieldDef<string>, Value>,
+  markShape: SymbolShape
+): SymbolShape {
+  if (channel !== 'shape') {
+    // use the value from the shape encoding or the mark config if they exist
+    const shape = getFirstConditionValue(shapeChannelDef) || markShape;
+    if (shape) {
+      return shape as SymbolShape;
```

Comment:
What if shape is a custom shape? (Why do we need to cast this as Symbol shape here?)

Reply:
SymbolShape can already be a string. 

```ts
export type SymbolShape =
  | 'circle'
  | 'square'
  | 'cross'
  | 'diamond'
  | 'triangle-up'
  | 'triangle-down'
  | 'triangle-right'
  | 'triangle-left'
  | 'arrow'
  | 'triangle'
  | 'wedge'
  | 'stroke'
  | string;
```

I had to make a few functions generics to remove the `as` here.

Reply:
See 75ffb4d27919f73ebf3ee50e4a65a8c1f052b590